utils/bundleHelper.js

```diff
@@ -3,14 +3,15 @@ import * as fs from "fs";
 
 let filePath = process.argv[2];
 let outputFileName = process.argv[3];
-$RefParser.bundle(filePath).then((schema) => {
+$RefParser.bundle(filePath).then(async (schema) => {
     let jsonSchema = JSON.stringify(schema, null, 2);
     jsonSchema = jsonSchema.replaceAll('%24', '$');
+    jsonSchema = jsonSchema.replaceAll('(%5E%5Ba-zA-Z%5D%5Ba-zA-Z0-9_%5D*%5Ba-zA-Z0-9%5D$)%7C(%5E%5Ba-zA-Z%5D$)', '(^[a-zA-Z][a-zA-Z0-9_]*[a-zA-Z0-9]$)|(^[a-zA-Z]$)');
```

Comment:
The issue lies with the percent-encoding (URI encoding) of these symbols, which as you noticed already happens with the dollar sign. Rather than replacing a specific text block (and then potentially having the same problem again if we change it or add new symbols), I think it would be better to decode all percent-encoded symbols. You could do it with separate calls to replace occurrences of `%5B` with `[`, `%5D` with `]`, `%5E` with `^` and `%7C` with `*`. Or, I think you can just use the built-in function `decodeURI()` on the whole jsonSchema string...

```suggestion
    let jsonSchema = decodeURI(JSON.stringify(schema, null, 2));
```

Reply:
I just tried with this decodeURI and seems like it won't work if we have the '$' case. Still gets the '%24'.

Reply:
I guess the dollar sign is allowed in URI encodings (like hashtag, question mark, etc). Still it's nice to have the general decoding for the majority of the characters!

Reply:
yeah I think so but it didn't work with vscode when I try the file with '%24' so that's why I still keep the replace there.